File: AEDS II/codigos/2 - palindromo/palindromoRecursivo.c
#include <stdio.h>
#include <string.h>

int recursaoPalindromo(char palavra[], int inicio, int fim){
    int ePalindromo = 1;
    if (inicio < fim){
        if (palavra[inicio] != palavra[fim]){
            inicio = fim;
            ePalindromo = 0;
        }
        else{ ePalindromo = recursaoPalindromo(palavra, inicio+1, fim-1); }   
    }
    return ePalindromo;
}

void palindromo(char palavra[]){
    int tamanho = strlen(palavra);
    
    if (recursaoPalindromo(palavra, 0, tamanho - 1)){ printf("SIM\n"); }
    else{ printf("NAO\n"); }
}

int main(){
    char palavra[500]; // 499. Tem que ter um \n no final
    scanf(" %[^\n]", palavra);

    while (strcmp(palavra, "FIM") != 0){
        palindromo(palavra);
        scanf(" %[^\n]", palavra);
    }
    return 0;
}

File: AEDS II/trabalhos/TP01Q01 - Palindromo/PalindromoRecursivo.java
import java.util.Scanner;

public class PalindromoRecursivo {
    private static boolean palindromoRecursivo(String palavra, int inicio, int fim) {
        boolean ePalindromo = true;

        if (inicio < fim) { 
            if (palavra.charAt(inicio) != palavra.charAt(fim)){ ePalindromo = false; }
            else{ ePalindromo = palindromoRecursivo(palavra, inicio+1, fim-1);  }
        }
        return ePalindromo;
    }
    public static void palindromo(String palavra) {
        int tamanho = palavra.length();
        if (palindromoRecursivo(palavra, 0, tamanho-1)) { System.out.println("SIM"); }
        else{ System.out.println("NAO"); }
    }
    public static void main(String[] args) {
        Scanner teclado = new Scanner(System.in);
            String palavra = teclado.nextLine();
            while (!palavra.equals("FIM")) {
                palindromo(palavra);
                palavra = teclado.nextLine();
            }
        teclado.close();
    }
}
